catalog/registry/catalog-registry-cswinputtransformer/src/main/java/ddf/catalog/registry/converter/RegistryPackageConverter.java

```diff
@@ -0,0 +1,652 @@
+        }
+    }
+
+    private static void parseRegistryPerson(PersonType person, RegistryMetacardImpl metacard)
+            throws RegistryConversionException {
+
+        validateIdentifiable(person);
+
+        Attribute pointOfContactAttribute = metacard.getAttribute(Metacard.POINT_OF_CONTACT);
+        if (pointOfContactAttribute == null || person.getId()
+                .contains(":primary")) {
+
+            String name = "no name";
+            String phone = "no telephone number";
+            String email = "no email address";
+
+            if (person.isSetPersonName()) {
+                PersonNameType personName = person.getPersonName();
+                List<String> nameParts = new ArrayList<>();
+                if (StringUtils.isNotBlank(personName.getFirstName())) {
+                    nameParts.add(personName.getFirstName());
+                }
+                if (StringUtils.isNotBlank(personName.getLastName())) {
+                    nameParts.add(personName.getLastName());
+                }
+
+                if (CollectionUtils.isNotEmpty(nameParts)) {
+                    name = String.join(" ", nameParts);
+                }
+            }
+
+            if (person.isSetTelephoneNumber()) {
+                List<TelephoneNumberType> phoneNumbers = person.getTelephoneNumber();
+                if (CollectionUtils.isNotEmpty(phoneNumbers)) {
+                    phone = getPhoneNumberString(phoneNumbers.get(0));
+                }
+            }
+
+            if (person.isSetEmailAddress()) {
+                List<EmailAddressType> emailAddresses = person.getEmailAddress();
+
+                if (CollectionUtils.isNotEmpty(emailAddresses)) {
+                    EmailAddressType emailAddress = emailAddresses.get(0);
+
+                    if (StringUtils.isNotBlank(emailAddress.getAddress())) {
+                        email = emailAddress.getAddress();
+                    }
+                }
+            }
+
+            String metacardPoc = String.format("%s, %s, %s", name, phone, email);
+            metacard.setAttribute(Metacard.POINT_OF_CONTACT, metacardPoc);
+        }
+    }
+
+    private static void parseRegistryService(ServiceType service, RegistryMetacardImpl metacard)
+            throws RegistryConversionException {
+
+        validateIdentifiable(service);
+
+        String xmlServiceBindingsTypesAttributeName = METACARD_XML_NAME_MAP.get(
+                RegistryObjectMetacardType.SERVICE_BINDING_TYPES);
+        String xmlServiceBindingsAttributeName = METACARD_XML_NAME_MAP.get(
+                RegistryObjectMetacardType.SERVICE_BINDINGS);
+
+        List<String> serviceBindings = new ArrayList<>();
+        List<String> serviceBindingTypes = new ArrayList<>();
+
+        for (ServiceBindingType binding : service.getServiceBinding()) {
+
+            List<String> bindings = new ArrayList<>();
+            List<String> bindingTypes = new ArrayList<>();
```

Comment:
This could potentially create a ton of these arraylists and kill performance when it comes to garbage collection. From looking at the code it looks like you could continue to use the same arraylists and simply clear them on every iteration as opposed to creating new lists every time.